sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuShuffleExchangeExec.scala

```diff
@@ -53,6 +53,16 @@ class GpuShuffleMeta(
     // when AQE is enabled and we are planning a new query stage, we need to look at meta-data
     // previously stored on the spark plan to determine whether this exchange can run on GPU
     wrapped.getTagValue(gpuSupportedTag).foreach(_.foreach(willNotWorkOnGpu))
+
+    if (shuffle.outputPartitioning.isInstanceOf[RoundRobinPartitioning] &&
```

Comment:
I think this would be better in the RoundRobinPartitioning rule in GpuOverrides, unless there is a reason it cannot work there.

Reply:
I don't think RoundRobinPartitioning has access to the input or output types.  It only takes a number of partitions argument, not the types being partitioned.  If there is a way to do it with the round-robin partition override, I agree it would be a much better place.

Reply:
That's right.